### ISSUE TITLE
Do we need special loss/optimizer when tfp layers involved? Can you use hybrid with non-probabilistic layers?

### ISSUE THREAD

**Issue by ben-arnao** [tensorflow/probability#822]
tfp layers seem interesting from the standpoint of generalizing possibly better than regular NN with dropout, batch norm, regularization, etc. I'm not as interested in getting a confidence values for my output.

Given that's the case do we need special loss/optimizer for the network if the last layer is not probabilistic?

Also can we combine regular layers with tfp layers?

ie.

```
tfp.DenseVariational
tf.Dropout
# i don't care about my biases being probabilistic
tf.Activation('relu')
```
